### ISSUE TITLE
How to customize ThreadEnforcer

### ISSUE THREAD

**Issue by chokhmah** [square/otto#190]
<img width="576" alt="screen shot 2016-08-12 at 12 07 13 pm" src="https://cloud.githubusercontent.com/assets/6762476/17612519/62b987a2-6085-11e6-8048-7faafee57d02.png">

Thanks.

**Comment by JakeWharton** [comment]
The default behavior verifies that the main thread is used, so you can just call `new Bus()`.

You can see the implementation of the main thread enforcer here: https://github.com/square/otto/blob/a85ae721d337d9129eb49f5dc90c8e65d49d48ca/otto/src/main/java/com/squareup/otto/ThreadEnforcer.java#L43-L50. A similar approach would be used for other threads as well.

**Comment by chokhmah** [comment]
Thanks.

My goal is to ensure that the Bus#post() will be run on main thread, because there are some method/class that calls Bus#post on a background thread. I used ThreadEnforcer.ANY and ThreadEnforcer.MAIN but it throws an exception when then on calling is from different thread.  Also The bus I created is a singleton.
My current implementation is I put Bus#post() inside a Handler#post() to make sure it is on the main thread. I am also thinking if I can customize the ThreadEnforcer to run the Bus#post on the main thread so that I won't use the Handler#post in handling the Bus#post.
Does my current implementation correct? or am I overthinking the approach on ThreadEnforcer?
